src/device/intrinsics/math.jl

```diff
@@ -240,9 +249,12 @@ using SpecialFunctions
 # NOTE: CUDA follows fmod, which behaves differently than Base.mod for negative numbers
 #@device_override Base.mod(x::Float64, y::Float64) = ccall("extern __nv_fmod", llvmcall, Cdouble, (Cdouble, Cdouble), x, y)
 #@device_override Base.mod(x::Float32, y::Float32) = ccall("extern __nv_fmodf", llvmcall, Cfloat, (Cfloat, Cfloat), x, y)
+@device_override Base.mod(x::Float16, y::Float16) = Float16(mod(Float32(x), Float32(y)))
```

Comment:
Why prefer this above the Base implementation?

Reply:
You're right - the `Base` version works here. I just removed this.